fhir-database-utils/src/test/java/com/ibm/fhir/database/utils/query/SearchQueryTest.java

```diff
@@ -0,0 +1,423 @@
+                on("comp2", "LOGICAL_RESOURCE_ID").eq("comp1", "LOGICAL_RESOURCE_ID")
+                .and("comp2", "PARAMETER_NAME_ID").eq(21329)
+                .and("comp2", "COMPOSITE_ID").eq("comp1", "COMPOSITE_ID")
+                .and().leftParen().col("comp2", "STR_VALUE").like(bind(likeValue)).escape("+").rightParen()
+                    )
+            .where("comp1", "LOGICAL_RESOURCE_ID").eq("LR1", "LOGICAL_RESOURCE_ID")
+            .and("comp1", "PARAMETER_NAME_ID").eq(21328)
+            .and().leftParen().col("comp1", "COMMON_TOKEN_VALUE_ID").eq(4464).rightParen()
+            .build();
+
+        Select ex1 = Select.select("1")
+            .from("Encounter_TOKEN_VALUES_V", alias("P1"))
+            .innerJoin("Observation_LOGICAL_RESOURCES", alias("LR1"),
+                on("LR1", "LOGICAL_ID").eq("P1", "TOKEN_VALUE")
+                .and("LR1", "VERSION_ID").eq().coalesce(col("P1", "REF_VERSION_ID"), col("LR1", "VERSION_ID"))
+                .and("P1", "PARAMETER_NAME_ID").eq(1149)
+                .and("P1", "CODE_SYSTEM_ID").eq(20129)
+                .and(isDeleted("LR1"))
+                    )
+            .where("P1", "LOGICAL_RESOURCE_ID").eq("LR0", "LOGICAL_RESOURCE_ID")
+            .and().leftParen().exists(ex2).rightParen()
+            .build();
+
+        Select main = Select.select("COUNT(*)")
+            .from("Encounter_LOGICAL_RESOURCES", alias("LR0"))
+            .where(isDeleted("LR0"))
+            .and().exists(ex1)
+            .build();
+
+        final List<BindMarkerNode> bindMarkers = new ArrayList<>();
+        StringStatementRenderer renderer = new StringStatementRenderer(TRANSLATOR, bindMarkers, false);
+        final String rendered = main.render(renderer);
+        assertEquals(rendered, SQL);
+
+        // Check we get the full list of bind markers
+        assertEquals(bindMarkers.size(), 1);
+
+        BindMarkerNode bindMarker = bindMarkers.get(0);
+        BindMarkerNodeVisitor v = new BindMarkerNodeVisitor() {
+
+            @Override
+            public void bindString(String value) {
+                assertEquals(value, likeValue);
+            }
+
+            @Override
+            public void bindLong(Long value) {
+                // not expected
+                assertFalse(true);
+            }
+
+            @Override
+            public void bindInt(Integer value) {
+                // not expected
+                assertFalse(true);
+            }
+
+            @Override
+            public void bindInstant(Instant value) {
+                // not expected
+                assertFalse(true);
+            }
+
+            @Override
+            public void bindDouble(Double value) {
+                // not expected
+                assertFalse(true);
+            }
+
+            @Override
+            public void bindBigDecimal(BigDecimal value) {
+                // not expected
+                assertFalse(true);
+            }
+        };
+        bindMarker.visit(v);
```

Comment:
What happens when there are no bind  markers?  it falls through? 
I've seen the code and  not super concerned, it's just worth confirming it ran when doing something like this.

Reply:
It can't fall through because we assert that the bindMarker size is 1 earlier.